internal/libyaml/api_test.go

```diff
@@ -0,0 +1,527 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package libyaml
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	"go.yaml.in/yaml/v4/internal/testutil/assert"
+)
+
+func TestAPI(t *testing.T) {
+	RunTestCases(t, "api_test.yaml", map[string]TestHandler{
+		"api-new":       runAPINewTest,
+		"api-method":    runAPIMethodTest,
+		"api-panic":     runAPIPanicTest,
+		"api-delete":    runAPIDeleteTest,
+		"api-new-event": runAPINewEventTest,
+	})
+}
+
+func runAPINewTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+	runFieldChecks(t, obj, tc.Checks)
+}
+
+func runAPIMethodTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+
+	// Run setup if any
+	if tc.Setup != nil {
+		if setupList, ok := tc.Setup.([]interface{}); ok && len(setupList) > 0 {
+			callMethodFromList(obj, setupList, tc.Bytes)
+		}
+	}
+
+	// Call the main method
+	callMethodFromList(obj, tc.Method, tc.Bytes)
+
+	// Run checks
+	runFieldChecks(t, obj, tc.Checks)
+}
+
+func runAPIPanicTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	obj := createObject(tc.Constructor)
+
+	// Run setup if any
+	if tc.Setup != nil {
+		if setupList, ok := tc.Setup.([]interface{}); ok && len(setupList) > 0 {
+			callMethodFromList(obj, setupList, tc.Bytes)
+		}
+	}
+
+	// The main method call should panic
+	// Want can be either a string or a single-element sequence
+	var wantMsg string
+	switch v := tc.Want.(type) {
+	case string:
+		wantMsg = v
+	case []interface{}:
+		msg, ok := v[0].(string)
```

Comment:
A nil/empty slice would panic

This pattern can be found at various place. They should all be addressed

Reply:
Fixed in 69894a0